### PR TITLE
282 WIP - remove linter warnings

### PR DESCRIPTION
**Pull request by zenyuk**
work in progress, don't merge

### CODE REVIEW

File: pkg/dhtring/dhtring.go [hunk 1]
```diff
@@ -355,8 +389,11 @@ func (r *Ring) get(hex string) *ringNode {
 	if r.size == 0 {
 		return nil
 	}
-	current := r.head
-	for ok := true; ok; ok = ((current != nil) && (current.val != r.head.val)) {
+	if r.head.val == hex {
```
Comment (zenyuk) [inline pkg/dhtring/dhtring.go:392]: there are new unit tests for this change below

File: pkg/fcrcrypto/hash.go [hunk 2]
```diff
@@ -37,8 +37,8 @@ func GetBlockchainHasher() hash.Hash {
 
 // BlockchainHash message digests some data using the algorithm used by the Filecoin blockchain.
 func BlockchainHash(data []byte) []byte {
-	hash := blake2b.Sum256(data)
-	return hash[:]
+	hashSum := blake2b.Sum256(data)
```
Comment (zenyuk) [inline pkg/fcrcrypto/hash.go:40]: name collision with imported package

File: pkg/fcrcrypto/keypair_retrieval.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func (k *KeyPair) EncodePrivateKey() string {
 // if the key algorithm is anything other than SigAlgEcdsaSecP256K1Blake2b.
 func (k *KeyPair) EncodeRawPrivateKey() (string, error) {
 	if k.alg.IsNot(SigAlgEcdsaSecP256K1Blake2b) {
-		return "", fmt.Errorf("Can not raw encode private key with algorithm: %d", k.alg.EncodeSigAlg())
+		return "", fmt.Errorf("can not raw encode private key with algorithm: %d", k.alg.EncodeSigAlg())
```
Comment (zenyuk) [inline pkg/fcrcrypto/keypair_retrieval.go:54]: errors should not start with a capital letters

File: pkg/fcrp2pserver/server.go [hunk 4]
```diff
@@ -119,7 +119,11 @@ func (s *FCRP2PServer) Start() error {
 // handleIncomingConnection handles incomming connection using given handlers.
 func (s *FCRP2PServer) handleIncomingConnection(conn net.Conn, handlers map[int32]func(reader *FCRServerReader, writer *FCRServerWriter, request *fcrmessages.FCRMessage) error) {
 	// Close connection on exit.
-	defer conn.Close()
+	defer func() {
```
Comment (zenyuk) [inline pkg/fcrp2pserver/server.go:122]: handling an Error returned by the Close() method

File: pkg/dhtring/dhtring.go [hunk 5]
```diff
@@ -148,12 +148,43 @@ func (r *Ring) Remove(hex string) {
...
 	}
-	current := r.head
-	for ok := true; ok; ok = current != nil && current.val != r.head.val {
+	if r.size == 1 {
+		r.head = nil
+		r.size = 0
+		return
+	}
+	if r.size == 2 {
+		if r.head.val == hex {
```
Comment (zenyuk) [inline pkg/dhtring/dhtring.go:160]: sorry there is a missing part for this. will be updated now

Comment (zenyuk) [reply pkg/dhtring/dhtring.go:160]: updated

**Review by wcgcyx — APPROVED** [review]

**Pull request merged**